python/ray/train/lightning/lightning_trainer.py

```diff
@@ -0,0 +1,108 @@
+            of ``pytorch_lightning.LightningModule``. This class should define your
+            model logic.
+        lightning_module_init_config: Configurations to pass into
+            ``lightning_module.__init__`` as kwargs.
+        trainer_init_config: Configurations to pass into
+            ``pytorch_lightning.Trainer.__init__`` as kwargs. For valid arguments to
+            pass, see
+            https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#init.
+        ddp_strategy_init_config: Configurations to pass into
+            ``pytorch_lightning.strategies.DDPStrategy.__init__`` as kwargs. Most users
+            should only set this to ``{"find_unused_parameters": False}`` or leave this
+            as-is. For valid arguments to pass, see
+            https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.strategies.DDPStrategy.html#pytorch_lightning.strategies.DDPStrategy
+            and
+            https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html#torch.nn.parallel.DistributedDataParallel.
+        torch_config: Configuration for setting up the PyTorch backend. If set to
+            None, use the default configuration. This replaces the ``backend_config``
+            arg of ``DataParallelTrainer``. Same as in ``TorchTrainer``.
+        scaling_config: Configuration for how to scale data parallel training.
+        dataset_config: Configuration for dataset ingest.
+        run_config: Configuration for the execution of the training run.
+        datasets: Any Ray Datasets to use for training. Use
+            the key "train" to denote which dataset is the training
+            dataset and (optionally) key "evaluation" to denote the evaluation
+            dataset. Can only contain a training dataset
+            and up to one extra dataset to be used for evaluation.
+            If a ``preprocessor`` is provided and has not already been fit,
+            it will be fit on the training dataset. All datasets will be
+            transformed by the ``preprocessor`` if one is provided.
+        preprocessor: A ray.data.Preprocessor to preprocess the
+            provided datasets.
+        resume_from_checkpoint: A checkpoint to resume training from.
+    """
+
+    def __init__(
+        self,
+        lightning_module: Type[pytorch_lightning.LightningModule],
+        *,
+        lightning_module_init_config: Optional[Dict] = None,
+        trainer_init_config: Optional[Dict] = None,
+        ddp_strategy_init_config: Optional[Dict] = None,
+        torch_config: Optional["TorchConfig"] = None,
+        scaling_config: Optional["ScalingConfig"] = None,
+        dataset_config: Optional[Dict[str, "DatasetConfig"]] = None,
+        run_config: Optional["RunConfig"] = None,
+        datasets: Optional[Dict[str, "GenDataset"]] = None,
+        preprocessor: Optional["Preprocessor"] = None,
+        resume_from_checkpoint: Optional["Checkpoint"] = None,
```

Comment:
So one observation is that there are a lot of configs in the current API...these are all needed, but there might be a middle ground API with less configs that need to be passed in, but also not requiring the user to setup the LightningModule and Trainer entirely by themselves.

cc @Yard1 for thoughts.

Reply:
I don't think we should try and come up with our own APIs here. It's better to delegate as much of the interface to PTL for both familiarity and ease of maintenance. If the user has to specify just as much as they would need to with PTL then that's fine IMO